### PR TITLE
Feature/1183/231 system test fixture

### PR DESCRIPTION
**Pull request by algattik**
## What this PR changes/adds

This the first of 2 PRs to add Azure transfer end-to-end test. The functionality has been split into 2 PRs to simplify the review process.

This first PR does not add any Azure transfer-specific test logic. It only restructures and improves the existing transfer test to make it extensible. The current file transfer Gatling test was split into a generic abstract class and a specific class for the File transfer logic.

The next PR will add another test subclass for Blob transfer.

## Further notes

Also improved Gatling to perform one additional request if the transfer successful, and validate that the transfer actually succeeded.

## Linked Issue(s)

Relates to #231

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] added relevant details to the changelog? (_skip with label `no-changelog`_)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [styleguide](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/styleguide.md) for details_)

### CODE REVIEW

File: system-tests/azure-tests/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/*
```
Comment (cpeeyush) [inline system-tests/azure-tests/build.gradle.kts:1]: This new module is not added in `settings.gradle.kts`

File: system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+        var actualBlobContent = destinationBlob.downloadContent().toString();
+        assertThat(actualBlobContent)
+                .withFailMessage("Transferred file contents are not same as the source file")
+                .isEqualTo(blobContent);
+    }
+
+    private String getProvisionedContainerName() {
+        return given()
+                .baseUri(CONSUMER_CONNECTOR_MANAGEMENT_URL + CONSUMER_MANAGEMENT_PATH)
+                .log().all()
```
Comment (cpeeyush) [inline system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java:157]: nit: logging can be removed.
```suggestion
```

File: system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                "contractPolicyId", policyId,
+                "criteria", criteria.getCriteria()
+        );
+
+        seedProviderData(CONTRACT_DEFINITIONS_PATH, contractDefinition);
+    }
+
+    private void seedProviderData(String path, Object requestBody) {
+        givenProviderBaseRequest()
+                .log().all()
```
Comment (cpeeyush) [inline system-tests/azure-tests/src/test/java/org/eclipse/dataspaceconnector/system/tests/local/BlobTransferIntegrationTest.java:225]: nit: logs
```suggestion
```

File: system-tests/runtimes/azure-storage-transfer-provider/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,67 @@
+/*
```
Comment (cpeeyush) [inline system-tests/runtimes/azure-storage-transfer-provider/build.gradle.kts:1]: Module not added in settings.gradle.kts.

File: system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/GatlingUtils.java [hunk 5]
```diff
@@ -40,8 +38,8 @@ public static void runGatling(Class<? extends Simulation> simulation, String des
 
         var statusCode = Gatling.fromMap(props.build());
 
-        assertThat(statusCode)
-                .withFailMessage("Gatling Simulation failed")
-                .isEqualTo(0);
+        if (statusCode != 0) {
+            throw new AssertionError("Gatling Simulation failed");
+        }
```
Comment (cpeeyush) [inline system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/GatlingUtils.java:43]: nit: just want to understand, why this change ?

File: system-tests/runtimes/azure-storage-transfer-consumer/build.gradle.kts [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*
```
[the remark this answers is not in the document]
Comment (algattik) [reply system-tests/runtimes/azure-storage-transfer-consumer/build.gradle.kts:1]: removed azure-tests (added by mistake)

File: system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/GatlingUtils.java (hunk 5 above)
Comment (algattik) [reply system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/GatlingUtils.java:43]: Not to take a dependency to assertj in an extensible module just for this.

**Review by cpeeyush — APPROVED** [review]

**Review by pkirch — APPROVED** [review]
Using `./gradlew clean build` I was able to build and run tests (as described in the readme).

Could not find any obvious concerns.

As I'm new to the project feel free to use my approval or to wait for another one. :-)

File: system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/GatlingUtils.java (hunk 5 above)
Comment (cpeeyush) [reply system-tests/tests/src/testFixtures/java/org/eclipse/dataspaceconnector/system/tests/utils/GatlingUtils.java:43]: 👍 okay
